### PR TITLE
Updates packages and typescript

### PR DESCRIPTION
**Pull request by lramos15**
Noted there was many security risks according to the npm audit.

1. Fixed security issues
2. Removed deprecated packages
3. Bumped the typescript version.

Had to fix the error typings. This was just a quick patch but believe we should probably use a type guard at some point.
Also had to upgrade the mongodb version which was a breaking change but I feel I updated it to use the proper new database functions.

### CODE REVIEW

File: author-verified/AuthorVerified.test.ts [hunk 1]
```diff
@@ -84,7 +84,7 @@ describe('AuthorVerified', () => {
 		expect(comments[0].body).to.contain('plz verify thx')
 	})
 
-	it.only('Unlocks the issue', async () => {
```
Comment (JacksonKearl) [inline author-verified/AuthorVerified.test.ts:87]: hah

File: classifier-deep/apply/apply-labels/index.ts [hunk 2]
```diff
@@ -54,11 +54,13 @@ class ApplyLabels extends Action {
 
 		if (manifestDbConnectionString) {
 			safeLog('has manifestDbConnectionString')
-			manifest = mongodb.MongoClient.connect(manifestDbConnectionString).then(async (db) => {
+			manifest = mongodb.MongoClient.connect(manifestDbConnectionString).then(async (client) => {
```
Comment (JacksonKearl) [inline classifier-deep/apply/apply-labels/index.ts:57]: I believe the new mongo client package is incompatible with our vscode-tools website's mongo version, I think we should just keep the old version. You can test with the connection string at https://ms.portal.azure.com/#@microsoft.onmicrosoft.com/resource/subscriptions/9b16b0df-65b6-46d3-b9c9-e4fbc86b1fa7/resourceGroups/vscode-tools/providers/Microsoft.DocumentDb/databaseAccounts/vscode-tools/mongoDbKeys

Comment (lramos15) [reply classifier-deep/apply/apply-labels/index.ts:57]: Updated the connection string and tested.

**Comment by lramos15** [comment]
Merging since main stipulation seemed to be mongo and that has now been tested against the database.

**Pull request merged**